File: src/main/java/ru/falseteam/neural2048/players/Player.java
package ru.falseteam.neural2048.players;

import ru.falseteam.neural2048.logic.GameLogic;

/**
 * Интерфейс от которого должен наследоваться любой алгоритм игры.
 *
 * @author Vladislav Sumin
 * @version 1.0
 */
public interface Player {
    /**
     * Игрок играет одну игру.
     */
    void playOneGame(GameLogic gameLogic);
}

File: src/main/java/ru/falseteam/neural2048/players/Player.kt
package ru.falseteam.neural2048.players

import ru.falseteam.neural2048.logic.GameLogic

/**
 * Интерфейс, от которого должен наследоваться любой алгоритм игры.
 *
 * @author Vladislav Sumin
 * @version 1.0
 */
abstract class Player(val gameLogic: GameLogic) {
    /**
     * Игрок играет одну игру.
     */
    abstract fun playOneGame()
}
